src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -829,3 +859,115 @@ void medViewContainer::removeColorIndicator(QColor color)
 {
     d->poolIndicator->removeColorIndicator(color);
 }
+
+/**
+ * @brief saves the scene in a XML file
+ * Saves views (all layers), and toolboxes parameters
+ * 	expected tree is as follow:
+ *  workingDir (user-defined)
+ * 		|-viewID0
+ * 			|-mapping.xml
+ * 			|-layerID01.xml
+ * 			|-layerID02.xml
+ * 			|-data_file (as saved by the dedicated writer)
+ */
+QString medViewContainer::saveScene()
+{
+    QString dirPath = QFileDialog::getExistingDirectory(this, tr("Open Directory"),"/home",QFileDialog::ShowDirsOnly);
+    QDir workingDir(dirPath);
+    QDomDocument doc("xml");
+    QDomElement root=doc.createElement("scene");
+    doc.appendChild(root);
+
+    QString subDirName;
+
+    medAbstractLayeredView* layeredView=dynamic_cast<medAbstractLayeredView*>(view());
+    if (layeredView!=0)
+    {
+        subDirName="view"+QUuid::createUuid ().toString();
+        if(!workingDir.mkdir(subDirName))
+            qWarning() << " failed to create new directory: "<<subDirName;
+        workingDir.cd(subDirName);
+        QString generatedPath=workingDir.canonicalPath()+"/mapping.xml";
+        QDomElement layeredViewInfo=doc.createElement("layeredView");
+        layeredViewInfo.setAttribute("path",subDirName+"/mapping.xml");
+        layeredViewInfo.setAttribute("id",layeredView->identifier());
+        root.appendChild(layeredViewInfo);
+        layeredView->write(generatedPath);
+        workingDir.cdUp();
+    }
```

Comment:
I think I've misread the code, not sure.

The condition I'm talking about is creating the folder, not the dynamic cast. 

The dynamic cast could always fail, yes - but practically it won't. That doesn't mean the code is safe, and as a general rule we should try to avoid problems like this. The issue is that the inheritance structure in medInria makes dynamic casts the easiest way to solve problems. I don't think there's much you can do in this case - there's no graceful way to fail from the dynamic_cast failure, because either the view is empty (which shouldn't happen), or a programmer has done something stupid, in which case a crash is warranted.

For the folder creation - we should definitely warn the user and abort the process if we can't create the folder. It's something the user can fix by elevating the privileges of the application, and we'll get unwanted behaviour if we continue.